code/src/components/Details.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react';
+import { useParams, Link } from 'react-router-dom';
+import { NotFound } from './NotFound';
+import '../css/Details.css';
+
+// This component shows movie details
+export const Details = () => {
+  // movieDetails stores movie information
+  const [movieDetails, setMovieDetails] = useState([])
+
+  // id helps us know which movie to show
+  const { id } = useParams()
+  // params is an object that holds values extracted from the URL,
+  // so we can use them in our code. In this case, it helps us get
+  // the movie ID from the URL.
+
+  // This function rounds the rating number
+  const roundedNumber = Math.round(movieDetails.vote_average * 10) / 10
+  // When we multiply movieDetails.vote_average by 10, we shift the decimal point
+  // one position to the right. Then, when we use Math.round(), we round the number
+  // we round the number to the nearest whole number. By doing this, we are
+  // effectively rounding the original value to one decimal place
+
```

Comment:
Good job with the commenting, it makes reading the code and your reasoning behind it so much easier!